core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/services/impl/OptionProviderServiceImpl.java

```diff
@@ -64,13 +83,25 @@ public List<Resource> getOptions(SlingHttpServletRequest request) {
         // Parse user-specified datasource settings from the request and/or underlying "datasource" resource,
         OptionSourceParameters parameters = OptionSourceParameters.forRequest(request);
 
-        // For each of the datasource paths, try retrieve a list of JCR-stored options
+        // For each of the datasource paths, try retrieve a list of JCR-stored options or if the datasource path is URL, try retrieve a list of options during the response of GET request
         for (PathParameters pathParameters : parameters.getPathParameters()) {
-            Resource datasourceResource = OptionSourceResolver.resolve(request, pathParameters.getPath(), pathParameters.getFallbackPath());
-            if (datasourceResource == null || datasourceResource instanceof NonExistingResource) {
-                continue;
+            String pathParameter = pathParameters.getPath();
+            if (isValidUrl(pathParameter)) {
```

Comment:
I'd suggest naming this `isUrl`. Solves one occasional problem -- how to write a Javadoc that does not look self-obvious and dumb, like `/** Does foo */ doFoo()` */ If your method is named isUrl, we can write down a true rightful Javadoc like "It makes sure that the provided string is formatted like an URL by trying to convert it to the URL object"
Moreover, we cannot be sure that the URL is valid because it may lead to a non-existent page

Reply:
done